hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+from django.db import models
+from mezzanine.pages.page_processors import processor_for
+from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+from mezzanine.pages.models import Page
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.contrib.contenttypes import generic
+
+# Create your models here.
+
+
+class OriginalFileInfo(AbstractMetaDataElement):
+
+    term = 'OriginalFileInfo'
+
+    fileTypeEnum=(  (None, 'Unknown'),
+                    ("SHP", "ESRI Shapefiles"),
+                    ("ZSHP", "Zipped ESRI Shapefiles"),
+                    ("KML", "KML"),
+                    ("KMZ", "KMZ"),
+                    ("GML", "GML"),
+                    ("SQLITE", "SQLite")
+                 )
+    fileType=models.TextField(max_length=128, choices=fileTypeEnum, default=None)
+    baseFilename=models.TextField(max_length=256, null=False, blank=False)
+    fileCount=models.IntegerField(null=False, blank=False, default=0)
+
+    filenameString = models.TextField(max_length=2048, null=True, blank=True)
+
+    class Meta:
+        # OriginalFileInfo element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    @classmethod
+    def create(cls, **kwargs):
+
+        for limit in ['fileType', 'baseFilename', 'fileCount']:
+            if not limit in kwargs:
+                raise ValidationError("For OriginalFileInfo meta, one or more attribute is missing.")
+
+        metadata_obj = kwargs['content_object']
+        ori_file_info = OriginalFileInfo.objects.create(fileType=kwargs['fileType'], baseFilename=kwargs['baseFilename'],
+                                                  fileCount=kwargs['fileCount'],
+                                                 content_object=metadata_obj,)
+        # save filenameString info
+        for key, value in kwargs.iteritems():
+            if key in ('filenameString'):
+                setattr(ori_file_info, key, value)
+                ori_file_info.save()
+
+        return ori_file_info
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        ori_file_info = OriginalFileInfo.objects.get(id=element_id)
+        if ori_file_info:
+            # save filenameString info
+            for key, value in kwargs.iteritems():
+                if key in ('fileType', 'baseFilename', 'fileCount', 'filenameString'):
+                    setattr(ori_file_info, key, value)
+                    ori_file_info.save()
+        else:
+            raise ObjectDoesNotExist("No OriginalFileInfo was found for the provided id: %d" % element_id)
```

Comment:
If you want to enforce that only those field names can be updated, then you could just make the parameters to `update()` explicitly named instead of allowing any named parameters in the first place with `**kwargs`. Or, If we want to move this entirely into the base class, just don't check the field names at all. It doesn't look like it should break anything to do otherwise, since you aren't raising an exception or anything if the fieldname is not allowed. It is just ignored silently, and that would continue to happen if you did nothing at all.

Is it required that the update invoke a real `save()` because of required save signals that might be important elsewhere? If you don't need them, then you could do this with an `update()` call on the manager, which does an SQL update without fetching the object in the first place, so it is more efficient.